src/block.js

```diff
@@ -119,9 +154,20 @@ Block.prototype.toBuffer = function (headersOnly) {
   writeInt32(this.version)
   writeSlice(this.prevHash)
   writeSlice(this.merkleRoot)
+  if (coins.isZcash(this.network)) {
+    writeSlice(this.finalSaplingRoot)
+  }
   writeUInt32(this.timestamp)
   writeUInt32(this.bits)
-  writeUInt32(this.nonce)
+  if (coins.isZcash(this.network)) {
+    writeSlice(this.nonce)
+    varuint.encode(this.solutionSize, buffer, offset)
+    offset += varuint.encode.bytes
+    writeSlice(this.solution)
+  } else {
+    // Not sure sure why the nonce is read as UInt 32 and not as a slice
```

Comment:
It's not being read, but written?

Reply:
Thanks. More important, do you know if the nonce is a 32 Bytes unsigned integer in bitcoin headers? If so, I'll just remove the comment